app/src/main/java/com/github/se/icebreakrr/MainActivity.kt

```diff
@@ -4,40 +4,65 @@ import android.os.Bundle
-import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.semantics.semantics
-import androidx.compose.ui.semantics.testTag
-import androidx.compose.ui.tooling.preview.Preview
-import com.github.se.icebreakrr.resources.C
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.navigation
+import androidx.navigation.compose.rememberNavController
+import com.github.se.icebreakrr.ui.navigation.NavigationActions
+import com.github.se.icebreakrr.ui.navigation.Route
+import com.github.se.icebreakrr.ui.navigation.Screen
+import com.github.se.icebreakrr.ui.sections.AroundYouScreen
+import com.github.se.icebreakrr.ui.sections.NotificationScreen
+import com.github.se.icebreakrr.ui.sections.ProfileScreen
 import com.github.se.icebreakrr.ui.theme.SampleAppTheme
 
 class MainActivity : ComponentActivity() {
+
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
-    setContent {
-      SampleAppTheme {
-        // A surface container using the 'background' color from the theme
-        Surface(
-            modifier = Modifier.fillMaxSize().semantics { testTag = C.Tag.main_screen_container },
-            color = MaterialTheme.colorScheme.background) {
-              Greeting("Android")
-            }
-      }
-    }
+    setContent { SampleAppTheme { Surface(modifier = Modifier.fillMaxSize()) { IcebreakrrApp() } } }
   }
 }
 
+/**
+ * Main composable function for the Icebreakrr app.
+ *
+ * This function sets up the navigation controller and defines the navigation graph for the app,
+ * including screens such as "Around You", "Profile", and "Notifications".
+ *
+ * @see AroundYouScreen
+ * @see ProfileScreen
+ * @see NotificationScreen
+ */
 @Composable
-fun Greeting(name: String, modifier: Modifier = Modifier) {
-  Text(text = "Hello $name!", modifier = modifier.semantics { testTag = C.Tag.greeting })
-}
+fun IcebreakrrApp() {
+  val navController = rememberNavController()
+  val navigationActions = NavigationActions(navController)
 
-@Preview(showBackground = true)
-@Composable
-fun GreetingPreview() {
-  SampleAppTheme { Greeting("Android") }
+  // TODO Implement Auth Screen navigation
+  NavHost(navController = navController, startDestination = Route.AROUND_YOU) {
+    navigation(
+        startDestination = Screen.AROUND_YOU,
+        route = Route.AROUND_YOU,
+    ) {
+      composable(Screen.AROUND_YOU) { AroundYouScreen(navigationActions) }
+    }
+
+    navigation(
+        startDestination = Screen.PROFILE,
+        route = Route.PROFILE,
+    ) {
+      composable(Screen.PROFILE) { ProfileScreen(navigationActions) }
+    }
+
+    navigation(
+        startDestination = Screen.NOTIFICATIONS,
+        route = Route.NOTIFICATIONS,
+    ) {
+      composable(Screen.NOTIFICATIONS) { NotificationScreen(navigationActions) }
+    }
+  }
```

Comment:
It isn't clear what "ProfileScreen" means, is it the screen of our own profile or of one we are seeind in the around you screen ?
Also, navigations are supposed to encapsulate feature-specific flows, so we could have the "around you screen" and the "other's profile screen" in the same navigation ?